// sub~account-recreate/js/api.js
"use strict";

var ajax = require( 'app~utils' ).ajax;

module.exports = {
  mixins: [
    ajax
  ],
  methods: {
    sendRecreateEmailAndPassword: function() {
      this.setFormIsRetrievingDataState( true );
      this.removeError();
      return this.$http.post( 'http://localhost:3000/users/' + encodeURI( this.user.id ) + '/restoreEmailAndPassword', {
                   data: {
                     email: this.user.newEmail,
                     password: this.user.newPassword,
                     token: this.user.tempToken
                   }
                 } )
                 .then( response => {
                   this.setFormIsRetrievingDataState( false );
                   this.setConfirmationEmailSent( true );
                   this.setUserNewPassword( '' );
                   this.setActiveForm('formEditEmail');
                 }, response => {
                   this.setFormIsRetrievingDataState( false );
                   this.handleErrorCase( response );
                 } );
    },
    sendEditEmail: function() {
      this.setFormIsRetrievingDataState( true );
      this.removeError();
      return this.$http.post( 'http://localhost:3000/users/' + encodeURI( this.user.id ) + '/editEmail', {
                   data: {
                     email: this.user.newEmail,
                     password: this.user.password
                   }
                 } )
                 .then( response => {
                   this.setFormIsRetrievingDataState( false );
                   this.setConfirmationEmailSent( true );
                 }, response => {
                   this.setFormIsRetrievingDataState( false );
                   this.handleErrorCase(response);
                 } );
    }
  }
};
