// sub~account-recreate/vuex/getters.js
"use strict";

module.exports = {

  isAccountRestored: function(state) {
    return state.states.isAccountRestored;
  },
  activeForm: function(state) {
    return state.activeForm;
  },
  isActiveFormValid: function(state) {
    //console.log(state.forms);
    return state.forms[state.activeForm].isValid;
  },
  userOrEmailIsRetrievingData: function(state) {
    return state.forms.formOldEmail.inputs.userOrEmail.isRetrievingData;
  },
  errorMessage: function(state) {
    return state.errorMessage;
  },
  isError: function(state) {
    return state.states.isError;
  },
  user: function(state) {
    return state.user;
  },
  isLandline: function(state) {
    return state.states.isLandline;
  },
  isMobile: function(state) {
    return state.states.isMobile;
  },
  isSmsCodeRequested: function(state) {
    return state.states.isSmsCodeRequested;
  },
  isUserOrEmailRetrievingData: function(state) {
    return state.forms.formOldEmail.inputs.userOrEmail.isRetrievingData;
  },
  isRequestSmsRetrievingData: function(state) {
    return state.forms.formRequestSms.isRetrievingData;
  },
  authyToken: function(state) {
    return state.forms.formVerify.inputs.verificationCode.value;
  },
  passportImage: function(state) {
    return state.user.passportImage;
  },
  isPassportSubmitted: function(state) {
    return state.states.isPassportSubmitted;
  },
  isImageLoaded: function(state) {
    return state.states.isImageLoaded;
  },
  isConfirmationEmailSent: function(state) {
    return state.states.isConfirmationEmailSent;
  }
};
