// sub~account-recovery/vuex/actions.js
"use strict";

var vuexActions = require('app~vuex').actions;

module.exports = {
  setInputValue: vuexActions.setInputValue,
  setInputValidationState: vuexActions.setInputValidationState,
  setFormValidationState: vuexActions.setFormValidationState,
  setInputIsRetrievingDataState: vuexActions.setInputIsRetrievingDataState,
  setFormIsRetrievingDataState: vuexActions.setFormIsRetrievingDataState,
  setLandline: vuexActions.setLandline,
  setMobile: vuexActions.setMobile,
  setError: vuexActions.setError,
  setSmsRequestedStateTrue: vuexActions.setSmsRequestedStateTrue,
  removeError: vuexActions.removeError,
  setUserId: vuexActions.setUserId,
  setPhonePart: vuexActions.setPhonePart,
  setAuthyId: vuexActions.setAuthyId,
  setActiveForm: vuexActions.setActiveForm,
  inputValueChanged: vuexActions.inputValueChanged,
  setPassportImage: vuexActions.setPassportImage,
  setPassportSubmitted: vuexActions.setPassportSubmitted,
  setImageLoaded: vuexActions.setImageLoaded,
  setConfirmationEmailSent: vuexActions.setConfirmationEmailSent,
  setTempToken: vuexActions.setTempToken
};
